engine/apps/telegram/models/connectors/channel.py

```diff
@@ -123,15 +123,23 @@ def send_alert_group_message(self, alert_group: AlertGroup) -> None:
                 chat_id=self.channel_chat_id, message_type=TelegramMessage.ALERT_GROUP_MESSAGE, alert_group=alert_group
             )
         except error.BadRequest as e:
-            if e.message == "Need administrator rights in the channel chat":
-                logger.warning(
-                    f"Could not send alert group to Telegram channel with id {self.channel_chat_id} "
-                    f"due to lack of admin rights. alert_group {alert_group.pk}"
-                )
-            elif e.message == "Chat not found":
+            if TelegramClient.error_message_is(
+                e,
+                [
+                    TelegramClient.BadRequestMessage.NEED_ADMIN_RIGHTS_IN_THE_CHANNEL,
+                    TelegramClient.BadRequestMessage.CHAT_NOT_FOUND,
+                ],
+            ):
+                if TelegramClient.error_message_is(
+                    e, [TelegramClient.BadRequestMessage.NEED_ADMIN_RIGHTS_IN_THE_CHANNEL]
+                ):
```

Comment:
Do we need this extra check/if block? Can't we just use the str representation of the error?

Reply:
good point 👍 I tried to keep the original wording of the error messaging, but probably not a big deal to update this (ie. `"lack of admin rights"` vs `"Need administrator rights in the channel chat"` (which is what `TelegramClient.BadRequestMessage.NEED_ADMIN_RIGHTS_IN_THE_CHANNEL` is))

updated in https://github.com/grafana/oncall/pull/4851/commits/02543461d889ce35235dd40f8c6dbd3b4f064979